keras/src/layers/layer.py

```diff
@@ -1629,20 +1655,14 @@ def compute_size(x):
                 output_size
                 and output_size > self._remat_mode.output_size_threshold
             ):
-                return remat.remat(layer_call)(*args, **kwargs)
+                return remat.remat(layer_call)
         elif self._remat_mode.mode == "activations":
             has_activation = (
                 hasattr(self, "activation") and self.activation is not None
             )
             if has_activation:
-                not_rematted_activation = self.activation
-                try:
-                    self.activation = remat.remat(not_rematted_activation)
-                    return layer_call(*args, **kwargs)
-                finally:
-                    self.activation = not_rematted_activation
-
-        return layer_call(*args, **kwargs)
+                return self.rematerialized_activation_call_wrapper(layer_call)
```

Comment:
You need to do declare a local wrapper like this (bound to layer_call):

```python
    @functools.wraps(layer_call)
    def rematerialized_activation_call_wrapper(*args, **kwargs):
      original_activation = self.activation
      rematted_activation = remat(original_activation)
      self.activation = rematted_activation
      res = layer_call(*args, **kwargs)
      self.activation = original_activation

    return rematerialized_activation_call_wrapper
```

Reply:
this would return a executed layer_call to https://github.com/keras-team/keras/blob/master/keras/src/ops/operation.py#L42 - which would error out again

Reply:
This is returning `rematerialized_activation_call_wrapper` not the result of it.

`rematerialized_activation_call_wrapper` is function.